libs/agent-sm/bootstrap/src/main/java/org/opensearch/javaagent/bootstrap/AgentPolicy.java

```diff
@@ -28,7 +28,92 @@ public class AgentPolicy {
     private static final Logger LOGGER = Logger.getLogger(AgentPolicy.class.getName());
     private static volatile Policy policy;
     private static volatile Set<String> trustedHosts;
-    private static volatile Set<String> classesThatCanExit;
+    private static volatile BiFunction<Class<?>, List<Class<?>>, Boolean> classesThatCanExit;
+
+    /**
+     * Everyone is allowed to call {@link System#exit} or {@link Runtime#halt}
+     */
+    public static final class AllCanExit implements BiFunction<Class<?>, List<Class<?>>, Boolean> {
+        /**
+         * AllCanExit
+         */
+        public AllCanExit() {}
+
+        /**
+         * Check if class is allowed to call {@link System#exit}, {@link Runtime#halt}
+         * @param caller caller class
+         * @param chain chain of call classes
+         * @return is class allowed to call {@link System#exit}, {@link Runtime#halt} or not
+         */
+        @Override
+        public Boolean apply(Class<?> caller, List<Class<?>> chain) {
+            return true;
+        }
+    }
+
+    /**
+     * Only caller is allowed to call {@link System#exit} or {@link Runtime#halt}
+     */
+    public static final class CallerCanExit implements BiFunction<Class<?>, List<Class<?>>, Boolean> {
```

Comment:
Should this be pulled into a test class if it only makes sense in the context of tests?